api/src/main/java/io/opentelemetry/common/AttributeKeyImpl.java

```diff
@@ -17,15 +17,24 @@
 package io.opentelemetry.common;
 
 import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
 
 @SuppressWarnings("rawtypes")
 @AutoValue
 abstract class AttributeKeyImpl<T> implements AttributeKey<T> {
 
   static <T> AttributeKeyImpl<T> create(String key, AttributeType type) {
-    return new AutoValue_AttributeKeyImpl<>(key, type);
+    return new AutoValue_AttributeKeyImpl<>(type, key);
   }
 
+  @Override
+  public String getKey() {
```

Comment:
But can't `key()` return null?

Reply:
It can, but I really want the public API to have the getKey() be non-nullable. We can't have the API blow up with null keys, but we really want the API to signal that it should never be null. I'm open to other suggestions to solve this one.

Reply:
Got it - yeah that's a corner case I've ran into before too. I wasn't sure that was the intention but makes sense to me. If we ever add nullness checking (want to!) we'll need some "nullness casts" for these cases, just as a future food for thought.

Reply:
Right now, we drop null keys, I believe, so this seems like a decent compromise. You can put null in there, but you'll never actually see a null-keyed attribute anywhere, since we throw them away.